org.eclipse.scanning.example/src/org/eclipse/scanning/example/scannable/MockNeXusSlit.java

```diff
@@ -11,92 +11,180 @@
+		super(name, d, level);
+	}
 
 	public MockNeXusSlit(String name, double d, int level, String unit) {
 		super(name, d, level, unit);
 	}
+	
+	public boolean isWritingOn() {
+		return writingOn;
+	}
 
-	public MockNeXusSlit(String name, double d, int level) {
-		super(name, d, level);
+	public void setWritingOn(boolean writingOn) {
+		this.writingOn = writingOn;
+	}
+
+	@ScanFinally
+	public void nullify() {
+		xLzSet   = null;
+		xLzValue = null;
+		yLzSet   = null;
+		yLzValue = null;
+	}
+
+	public NexusObjectProvider<NXslit> getNexusProvider(NexusScanInfo info) throws NexusException {
+		final NXslit positioner = NexusNodeFactory.createNXslit();
+
+		if (info.getScanRole(getName()) == ScanRole.METADATA) {
+			positioner.setX_gapScalar(getPosition().doubleValue());
+			positioner.setY_gapScalar(getPosition().doubleValue());
+		} else {
+			String floatFill = System.getProperty("GDA/gda.nexus.floatfillvalue", "NaN");
+			double fill = "NaN".equalsIgnoreCase(floatFill) ? Double.NaN : Double.parseDouble(floatFill);
+
+			xLzSet = positioner.initializeLazyDataset(NXslit.NX_X_GAP, 1, Double.class);
+			yLzSet = positioner.initializeLazyDataset(NXslit.NX_Y_GAP, 1, Double.class);
+			xLzSet.setFillValue(fill);
+			yLzSet.setFillValue(fill);
+			xLzSet.setChunking(new int[]{8}); // Faster than looking at the shape of the scan for this dimension because slow to iterate.
+			yLzSet.setChunking(new int[]{8}); // Faster than looking at the shape of the scan for this dimension because slow to iterate.
+			xLzSet.setWritingAsync(true);
+			yLzSet.setWritingAsync(true);
+
+			xLzValue = positioner.initializeLazyDataset(NXslit.NX_X_GAP, info.getRank(), Double.class);
+			yLzValue = positioner.initializeLazyDataset(NXslit.NX_Y_GAP, info.getRank(), Double.class);
+			xLzValue.setFillValue(fill);
+			yLzValue.setFillValue(fill);
+			xLzValue.setChunking(info.createChunk(false, 8)); // Might be slow, need to check this
+			yLzValue.setChunking(info.createChunk(false, 8)); // Might be slow, need to check this
+			xLzValue.setWritingAsync(true);
+			yLzValue.setWritingAsync(true);
+		}
+
+		registerAttributes(positioner, this);
+
+		NexusObjectWrapper<NXslit> nexusDelegate = new NexusObjectWrapper<>(
+				getName(), positioner, NXslit.NX_X_GAP);
+		nexusDelegate.setDefaultAxisDataFieldName(NXslit.NX_X_GAP);
+		nexusDelegate.setCategory(NexusBaseClass.NX_INSTRUMENT);
+		return nexusDelegate;
+	}
+
+	public void setPosition(Number value, IPosition position) throws Exception {
+
+		if (value!=null) {
+			int index = position!=null ? position.getIndex(getName()) : -1;
+			if (isRealisticMove()) {
+				value = doRealisticMove(value, index, -1);
+			}
+			this.position = value;
+			delegate.firePositionPerformed(-1, new Scalar(getName(), index, value.doubleValue()));
+		}
+
+		if (position!=null) {
+			write(value, getPosition(), position);
+		}
+	}
+
+	private void write(Number demand, Number actual, IPosition loc) throws Exception {

```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this unused private "write" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00112)